src/2024h2/cargo-script.md

```diff
@@ -0,0 +1,71 @@
+# cargo-scrpt
+
+| Metadata | |
+| --- | --- |
+| Owner(s) | epage |
+| Teams | Cargo |
+| Status | WIP |
+
+## Motivation
+
+Being able to have a Cargo package in a single file can reduce friction in development and communication,
+improving bug reports, educational material, prototyping, and development of small utilities.
+
+### The status quo
+
+Today, at minimum a Cargo package is at least two files (`Cargo.toml` and either `main.rs` or `lib.rs`).
+The `Cargo.toml` has several required fields.
+
+To share this in a bug report, people resort to
+- Creating a repo and sharing it
+- A shell script that cats out to multiple files
+- Manually specifying each file
+- Under-specifying the reproduction case (likely the most common due to being the eaisest)
+
+To create a utility, a developer will need to run `cargo new`, update the
+`Cargo.toml` and `main.rs`, and decide on a strategy to run this (e.g. a shell
+script in the path that calls `cargo run --manifest-path ...`).
+
+### The next few steps
+
+With [RFC 3502](https://github.com/rust-lang/rfcs/pull/3502)
+and [RFC 3503](https://github.com/rust-lang/rfcs/pull/3503)
+approved, the next steps are being tracked in [#12207](https://github.com/rust-lang/cargo/issues/12207).
+
+At a high-level, this is
+- Add support to the compiler for the frontmatter syntax
+- Add support in Cargo for scripts as a "source"
+- Polish
```

Comment:
Maybe I should rename this section, but it's meant to be *what do you aim to get done by EOY 2024 (in next 6 months)* -- is this the list?

Reply:
Yes, we've had a prototype in Cargo for over a year now.  While a lot is done at this point, the cross-team aspect is why I felt this could still be worth a team goal (plus that "polish" can cover quite a bit).

Reply:
OK!